sycl/test-e2e/Matrix/Inputs/joint_matrix_bfloat16_impl.hpp

```diff
@@ -6,53 +6,60 @@
+          size_t TK>
+class imatrix;
 
-template <typename T1, typename T2, size_t M, size_t N, size_t K, size_t TM,
-          size_t TN, size_t TK>
-void matrix_multiply(big_matrix<T1, M, N> &C, big_matrix<T2, M, K> &A,
-                     big_matrix<T2, K / 2, N * 2> &B) {
+template <typename T, typename TAcc, typename TResult, size_t M, size_t N,
+          size_t K, size_t TM, size_t TN, size_t TK>
+void matrix_multiply(big_matrix<TResult, M, N> &D, big_matrix<TAcc, M, N> &C,
+                     big_matrix<T, M, K> &A, big_matrix<T, K, N> &B) {
   size_t NDRangeM = M / TM;
   size_t NDRangeN = N / TN;
-  buffer<T2, 2> bufA(A.get_data(), range<2>(M, K));
-  buffer<T2, 2> bufB(B.get_data(), range<2>(K, N));
-  buffer<T1, 2> bufC((T1 *)C.get_data(), range<2>(M, N));
-
+  buffer<T, 2> bufA(A.get_data(), range<2>(M, K));
+  buffer<T, 2> bufB(B.get_data(), range<2>(K, N));
+  buffer<TAcc, 2> bufC((TAcc *)C.get_data(), range<2>(M, N));
+  buffer<TResult, 2> bufD((TResult *)D.get_data(), range<2>(M, N));
   queue q;
-  size_t sg_size = get_sg_size<imatrix<T1, TM, TN, TK>>(q);
+  size_t sg_size = get_sg_size<imatrix<T, TAcc, TResult, TM, TN, TK>>(q);
+
   q.submit([&](handler &cgh) {
      accessor accA{bufA, cgh};
      accessor accB{bufB, cgh};
      accessor accC{bufC, cgh};
+     accessor accD{bufD, cgh};
 
-     cgh.parallel_for<imatrix<T1, TM, TN, TK>>(
+     cgh.parallel_for<imatrix<T, TAcc, TResult, TM, TN, TK>>(
          nd_range<2>({NDRangeM, NDRangeN * sg_size}, {1, 1 * sg_size}),
          [=](nd_item<2> spmd_item)
 #ifdef SG_SZ
              [[sycl::reqd_sub_group_size(SG_SZ)]]
 #endif
          {
-           // The submatrix API has to be accessed by all the workitems in a
-           // subgroup these functions will be called once by the subgroup no
-           // code divergence between the workitems
+           //            // The submatrix API has to be accessed by all the
+           //            workitems in a
+           //            // subgroup these functions will be called once by the
+           //            subgroup no
+           //            // code divergence between the workitems
            const auto global_idx = spmd_item.get_global_id(0);
            const auto global_idy = spmd_item.get_global_id(1);
            const auto sg_startx = global_idx - spmd_item.get_local_id(0);
            const auto sg_starty = global_idy - spmd_item.get_local_id(1);
 
            sub_group sg = spmd_item.get_sub_group();
-           joint_matrix<sub_group, T2, use::a, TM, TK, layout::row_major> sub_a;
+           joint_matrix<sub_group, T, use::a, TM, TK, layout::row_major> sub_a;
            // For B, we assume B has been already VNNIed.
-           joint_matrix<sub_group, T2, use::b, TK, TN, layout::ext_intel_packed>
-               sub_b;
-           joint_matrix<sub_group, T1, use::accumulator, TM, TN> sub_c;
+           joint_matrix<sub_group, T, use::b, TK, TN, layout::row_major> sub_b;
```

Comment:
I'm wondering why did you change packed layout to row_major? Not this test is checking different set of APIs. Is this intended change?